client/src/features/users/Auth/Steps/CreateOrgForm.tsx

```diff
@@ -0,0 +1,293 @@
+import { Organization } from '../../../../types';
+import { useMutation, useQuery } from 'react-query';
+import axios, { AxiosResponse } from 'axios';
+import SimpleSnackbar from '../../../action/assets/SimpleSnackbar';
+import { green } from '@mui/material/colors';
+
+const classifications = [
+  { value: 'charitable', text: 'Charitable Organization' },
+  { value: 'religious', text: 'Religious Organization' },
+  { value: 'private', text: 'Private Foundation' },
+  { value: 'political', text: 'Political Organizations' },
+  { value: 'other', text: 'Other' },
+];
+
+const orgValidationSchema = Yup.object().shape({
+  doing_business_as: Yup.string().required('Required'),
+  city: Yup.string().required('Required'),
+  state: Yup.string().required('Required').min(2).max(2),
+  ein: Yup.string()
+    .matches(/^[0-9]\d?-\d{7}$/, 'EIN must match: 99-9999999')
+    .required('Required'),
+  name: Yup.string(),
+  description: Yup.string().required('Required'),
+  website: Yup.string().required('Required'),
+  address: Yup.string().required('Required'),
+  phone: Yup.string()
+    .matches(/^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]\d{3}[\s.-]\d{4}$/, 'EIN must match: 99-9999999')
+    .required('Required'),
+  nonprofit_classification: Yup.string().min(2).required('Required'),
+});
+
+interface Props {
+  triggerNextStep: (step: number) => void;
+  setParentOrg: (createdOrg: Organization) => void;
+  parentOrg: Organization;
+  classes: ClassNameMap<'button' | 'header' | 'arrow' | 'sideImg' | 'signUpContainer'>;
+}
+const CreateOrgForm = ({
+  setParentOrg = () => {},
+  parentOrg,
+  triggerNextStep = () => {},
+  classes,
+}: Props) => {
+  const [org, setOrg] = React.useState<Organization>(parentOrg);
+  const [triggerEinSearch, setTriggerEinSearch] = React.useState<boolean>(false);
+
+  const orgCreateMutation = useMutation<AxiosResponse<any, any>, Error, Organization, Error>(
+    (newOrg: Organization) => {
+      return axios.post(`http://localhost:3001/api/organizations`, { ...newOrg });
+    },
+    {
+      onSuccess: (data: any) => {
+        triggerNextStep(1);
+        setParentOrg((data as any).data);
+      },
+    },
+  );
+
+  const devNull = () => {};
+
+  const orgValidateEinQuery = useQuery<
+    AxiosResponse<any, any>,
+    unknown,
+    { ein: string; name: string },
+    string[]
+  >({
+    enabled: triggerEinSearch,
+    queryKey: ['orgValidateEinQuery', org.ein],
+    queryFn: ({ queryKey }) => {
+      const [, ein] = queryKey;
+      return axios.get(`http://localhost:3001/api/organizations/ein/${ein}`);
+    },
+    onSuccess: (res: any) => {
+      setTriggerEinSearch(false);
+      setOrg({ ...org, name: res.data.name });
+    },
+    onError: (res: any) => {
+      setOrg({ ...org, name: '' });
+    },
+    retry: 0,
+  });
+
+  const handleNext = (values: Organization) => {
+    if (org.id === undefined) {
+      setOrg({ ...org });
+      orgCreateMutation.mutate({ ...values, name: org.name });
```

Comment:
seems to still be making request to create a form on `next` click from step 1

<img width="1427" alt="Screen Shot 2022-06-17 at 21 49 59" src="https://user-images.githubusercontent.com/16400333/174423331-b74f43d5-8c57-4d02-ad8f-adb48ea9142d.png">

Reply:
i think `handleNext` should only do `triggerNextStep(1);` right?
well, and validate that all inputs are filled out